packages/utilities/adaptive-ui/src/color/recipe.ts

```diff
@@ -56,3 +58,58 @@ export interface InteractiveSwatchSet {
      */
     focus: Swatch;
 }
+
+export interface InteractiveColorTokens {
+    /**
+     * Interface recipe
+     */
+    recipe: DesignToken<InteractiveColorRecipe>;
+
+    set: DesignToken<InteractiveSwatchSet>;
+
+    rest: CSSDesignToken<Swatch>;
+
+    hover: CSSDesignToken<Swatch>;
+
+    active: CSSDesignToken<Swatch>;
+
+    focus: CSSDesignToken<Swatch>;
+}
+
+export function createInteractiveColorTokens(
+    recipe: (element: HTMLElement, reference?: Swatch) => InteractiveSwatchSet,
+    name: string
+): InteractiveColorTokens {
+    const recipeToken = createNonCss<InteractiveColorRecipe>(
+        `${name}-recipe`
+    ).withDefault({
+        evaluate: recipe,
+    });
+
+    const setToken = createNonCss<InteractiveSwatchSet>(`${name}-set`).withDefault(
+        (element: HTMLElement): InteractiveSwatchSet =>
+            recipeToken.getValueFor(element).evaluate(element)
+    );
+
+    return {
+        recipe: recipeToken,
+
+        set: setToken,
+
+        rest: create<Swatch>(`${name}-rest`).withDefault(
+            (element: HTMLElement) => setToken.getValueFor(element).rest
+        ),
+
+        hover: create<Swatch>(`${name}-hover`).withDefault(
+            (element: HTMLElement) => setToken.getValueFor(element).hover
+        ),
```

Comment:
Per below, would this property be better as a cached function, where the design tokens is not created until the first time it's referenced?

Reply:
Yeah, assigning a default value will effectively prevent any tree-shaking which may be undesirable.